### PR TITLE
changed cookie parsing method with suggestion from https://github.com…

### PR DESCRIPTION
**Pull request by Ross-s**
changed cookie parsing method with suggestion from https://github.com/BuilderIO/builder/issues/1679 and prevented undefined from being in final oputput if the sanitised version is undefined

## Description
As per https://github.com/BuilderIO/builder/issues/1679 the same issue was hapening for me, so i have changed the code to use the suggested method. I have also prevented undefined from being allowed in the outputed attributes as it is also possible for the sanitizedKey to be undefined

### CODE REVIEW

File: packages/personalization-utils/src/next.ts [hunk 1]
```diff
@@ -22,13 +22,15 @@ export function getPersonalizedURL(
...
   const query = Object.fromEntries(url.searchParams);
-  const allCookies = Array.from(request.cookies.entries()).reduce(
-    (acc, [key]) => ({
-      ...acc,
-      [key]: tryJsonParse(request.cookies.get(key)!),
-    }),
-    {}
-  );
+  const allCookies: Record<string, any> = {};
+    for (const cookie of request.cookies) {
```
Comment (teleaziz) [inline packages/personalization-utils/src/next.ts:26]: indentation, you can run from root:
```
yarn prettier -- write ./packages/personalization-utils/src/*.ts
```

**Review by teleaziz — COMMENTED** [review]
LGTM, thanks for the pr, left one comment

**Comment by Ross-s** [comment]
I have cleaned up the formating @teleaziz

**Review by teleaziz — APPROVED** [review]

**Comment by Ross-s** [comment]
strange, my pr has nothing to do with styling, but a styling unit test failed?

**Comment by samijaber** [comment]
@Ross-s These unit tests had some flakiness issues that were resolved last week. I merged the latest `main` back into your PR to address the issue.

**Comment by teleaziz** [comment]
changed released in `@builder.io/personalization-utils@2.0.1-0` thanks again @Ross-s , I'll make a major version release next as this will break for users on older next versions

**Pull request merged**